frontend/catalyst/pennylane_extensions.py

```diff
@@ -428,7 +438,85 @@ def circuit(x):
     >>> workflow(2.0)
     array(-3.14159265)
     """
-    return Grad(_ensure_differentiable(f), grad_params=_check_grad_params(method, h, argnum))
+    scalar_out = True
+    return Grad(
+        _ensure_differentiable(f), grad_params=_check_grad_params(method, scalar_out, h, argnum)
+    )
+
+
+def jacobian(f: DifferentiableLike, *, method=None, h=None, argnum=None):
+    """A :func:`~.qjit` compatible Jacobian transformation for PennyLane/Catalyst.
+
+    This function allows the Jacobian of a hybrid quantum-classical function
+    to be computed within the compiled program.
+
+    .. warning::
+
+        If parameter-shift or adjoint is specified, this will only be used
+        for internal _quantum_ functions. Classical components will be differentiated
+        using standard reverse-mode automatic differentiation.
```

Comment:
Is this warning still needed now that the allowed diff methods are `defer` and `fd`?